USE_CASES.md

```diff
@@ -127,3 +128,291 @@ Find more information about all of SendGrid's whitelabeling related documentatio
+```
+
+### SETUP MAILER
+
+#17
+
+Create a file in `config/initializers` named `setup_mail.rb`:
+
+```
+$ touch config/initializers/setup_mail.rb
+```
+add the following code:
+
+```ruby
+if Rails.env.development? || Rails.env.production?
+  ActionMailer::Base.delivery_method = :smtp
+  ActionMailer::Base.smtp_settings = {
+    address:        'smtp.sendgrid.net',
+    port:           '587',
+    authentication: :plain,
+    user_name:      ENV['SENDGRID_USERNAME'],
+    password:       ENV['SENDGRID_PASSWORD'],
+    domain:         'heroku.com',
+    enable_starttls_auto: true
+  }
+end
+```
+The code in `config/initialize` runs when our app starts. We need to configure some special settings to send emails.
+
+#18
+
+Lets, generate a mailer
+
+```ruby
+$ rails generate mailer HelloMailer
+```
+
+#19
+
+In your `Gemfile` add `gem sendgrid-ruby` and run `$ bundle update`
+
+#20
+
+In `app/mailers/hello_mailer.rb` add:
+
+```ruby
+require 'sendgrid-ruby'
+
+class HelloMailer < ApplicationMailer
+  include SendGrid
+
+  def self.new_message(email)
+
+    from = Email.new(email: 'replacethiswithyouremail@gmail.com') #TODO
+    subject = 'Hello World from the SendGrid Ruby Library!'
+    to = Email.new(email: email)
+    content = Content.new(type: 'text/plain', value: 'Hello, Email!')
+    mail = Mail.new(from, subject, to, content)
+
+    sg = SendGrid::API.new(api_key: ENV['SENDGRID_API_KEY'])
+    response = sg.client.mail._('send').post(request_body: mail.to_json)
+    puts response.status_code
+    puts response.body
+    puts response.headers
+
+  end
+end
+```
+
+This is a simple example, but if you would like to to use a SendGrid Template, have a look at this [example](https://github.com/sendgrid/sendgrid-ruby/blob/master/USE_CASES.md)
+
+### Testing
+
+We can test the mailer in the Rails console:
+
+```
+$ rails console
+2.3.3 :001 > HelloMailer.new_message('someone@gmail.com')
+```
+
+If everything is setup correctly, you should receive an email in your inbox within a couple of minutes.
+
+### Using the email method:
+
+You can use `HelloMailer.new_message()` in a controller or a model:
+
+For example if you want to send a Hello Email when a _new user_ is created:
+
+
+Call method from Controller:
+
+```ruby
+class UserController < ApplicationController
+  def create
+    @user = User.new(params[:user])
+
+    if @user.save
+      HelloMailer.new_message(@user.email)
+    end
+    # some code goes here....
+  end
+end
+```
+
+Or call method from Model:
+
+```ruby
+class User < ActiveRecord::Base
+  after_create :send_hello_email
+
+  private
+
+  def send_hello_email
+    HelloMailer.new_message(self.email)
+  end
+end
+```
+
+### Deploy to Heroku
+
+```
+$ git push heroku master
+```
+
+_Note: This example doesn't use a database, but if your application has a database then migrate your production database to Heroku_ - [instructions](https://devcenter.heroku.com/articles/getting-started-with-rails4#migrate-your-database)
```

Comment:
One final link to embiggen. Instead of:

> migrate your production database to Heroku_ - [instructions]

How about:

    [migrate your production database to Heroku]